### DISCUSSION TITLE
Variable (property) substitution in include file attributes no longer working in 1.4.8

### DISCUSSION THREAD

**Discussion by peterkempy** [qos-ch/logback#695] [General]
```
<configuration scan="true" scanPeriod="30 seconds">
    <variable name="application.id" value="my-application"/>
...
    <include optional="true" file="/mypath/${application.id}/config/logback-override.xml"/>

```
**_This_** substitution doesn't happen in this case (other substitutions are still fine)

This is what I get in the logs:

`11:46:28,122 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@333d111 - Adding [file:/mypath/application.id_IS_UNDEFINED/config/logback-override.xml] to configuration watch list.`

Anyone else had this issue ?
(Worked in previous versions of logback - I'll do some digging to find out exactly when it stopped working)

**Comment by ceki** [comment]
@peterkempy Thank you for this report. Please create a jira issue at jira.qos.ch so that this problem can be tracked.

**Comment by ceki** [comment]
Created https://jira.qos.ch/browse/LOGBACK-1760

**Comment by Mastermindaxe** [comment]
@ceki Facing the same issue. In the process of upgrading to Spring Boot 3 and this is breaking our code :( Were you able to find out in which version this regression came to be so we can downgrade? I'll try some versions out otherwise. Maybe I'll find it in the meantime

EDIT: Can confirm that this problem persists when using version `1.4.0`. That way we can't pin a version with Jakarta instead of javax.

**Comment by ceki** [comment]
Issue is fixed in commit af92be47  to be released in logback version 1.5.5
